packages/SwingSet/src/vats/vat-admin/vat-vat-admin.js

```diff
@@ -193,6 +207,17 @@ export function buildRootObject(vatPowers) {
     }
   }
 
+  // the kernel queues this to us when a vat upgrade completes or fails
+  function vatUpgradeCallback(upgradeID, success, error) {
+    const { resolve, reject } = pendingUpgrades.get(upgradeID);
+    pendingUpgrades.delete(upgradeID);
```

Comment:
Thinking back to the thing we had to do with the race between the `pending` and `running` maps in the event of vat termination during `startVat`, I'm wondering if there's possibly a similar kind of thing here with `pendingUpgrades`.  Haven't thought deeply about it but calling the notion to your attention just in case.

Reply:
That's a great point, if the vat is terminated while we've got an upgrade pending, what should we do? Also, if someone creates a vat, then schedules an upgrade right away before the `startVat` has even had a chance to run (this "continuous deployment" thing taken to an extreme), and then `startVat` fails.

I'm going to make a note to address this as the upgrade process matures. The answer might be to reject all `pendingUpgrades` if/when we get `vatTerminated`, or if we get a `newVatCallback` with an error.